velox/functions/prestosql/aggregates/tests/SetUnionTest.cpp

```diff
@@ -500,5 +500,98 @@ TEST_F(SetUnionTest, longDecimal) {
   testAggregations(
       {data}, {}, {"set_union(c0)"}, {"array_sort(a0)"}, {expected});
 }
+
+TEST_F(SetUnionTest, nans) {
+  // Verify that NaNs with different binary representations are considered equal
+  // and deduplicated.
+  static const auto kNaN = std::numeric_limits<double>::quiet_NaN();
+  static const auto kSNaN = std::numeric_limits<double>::signaling_NaN();
+
+  // Global aggregation, Primitive type.
+  auto data = makeRowVector({
+      makeArrayVector<double>({
+          {1, 2, kNaN},
+          {3, 4, kNaN},
+          {3, 2, kSNaN},
+      }),
+      makeFlatVector<int32_t>({1, 2, 2}),
+  });
+
+  auto expected = makeRowVector({
```

Comment:
I know kNan = kSNaN, but  is it possible to add another data source just like data , but with kSNaN in place of kNaN and also ensure its equal to expected.

Reply:
Do you mind clarifying with an example? There is a kSNaN in the input data at L515, would that suffice?

Reply:
So if we do : 

 auto data2 = makeRowVector({
      makeArrayVector<double>({
          {1, 2, kSNaN},
          {3, 4, kSNaN},
          {3, 2, kNaN},
      }),
      makeFlatVector<int32_t>({1, 2, 2}),
  });
  
  Then even for data2 
  ```
  testAggregations(
      {data2}, {}, {"set_union(c0)"}, {"array_sort(a0)"}, {expected});
     ```
    is correct.